### PR TITLE
Discover Safari and Mobile Compatibility

### LINKED ISSUE #3698: The icon for the user's account and to logout is missing for some users
For the user "ndebuhr", but not other test accounts, the account button in the top right to view profile, logout, etc. is not visible.  

[005.zip](https://github.com/hydroshare/hydroshare/files/3884657/005.zip)

### LINKED ISSUE #3926: Inconsistent capitalization of legend categories in Discover
The legend categories are not consistently capitalized.  Consider changing "Sharing and Access" to "Sharing and access".

### LINKED ISSUE #4033: Discover results table and facets do not resize responsively
Discover results table and facets do not resize responsively. Instead when the browser size decreases, the table is pushed to the bottom of the screen leaving a blank space that takes up most of the page. This table and facets should resize using responsive design similar to the "My Resources" page.

The "Discover" page elements do not resize when the page is resized: 
image

The "My Resources" page elements do resize when the page is resized:

image

### PR DESCRIPTION
**Pull request by cuahsimarko**
Verify this work at https://dev-hs-6.cuahsi.org/search/ using your production credentials.

Fixes #4033
Fixes #3698 (or can't reproduce)
Fixes #3926
Works on macOS Catalina Safari 13.x

Since extensive cleanup in the source code was performed, a keen eye on functionality to ensure there are no regressions is warranted.

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW VULNERABILITIES | 0 |
| NEW DUPLICATED LINES DENSITY | 10.8552631578947 |
| NEW CODE SMELLS | 0 |
| NEW SECURITY HOTSPOTS | 0 |
| NEW BUGS | 13 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4160](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4160)

File: hs_discover/src/components/Search.vue [hunk 1]
```diff
@@ -347,7 +369,7 @@ export default {
     if (this.subjectFilter) {
       const escaped = [];
       this.subjectFilter.forEach((x) => {
-        escaped.push(x.replaceAll('\\', '\\\\'));
+        escaped.push(x.replace(/\\/g, '\\\\'));
```
Comment (hyi) [inline hs_discover/src/components/Search.vue:372]: Nice to know the regex in the first parameter of the replace() does not need to be quoted as a string.

**Review by hyi — APPROVED** [review]
I am approving this PR with a comment that css styles can be managed as a separate css file rather than being embedding into html files and templates in vue files which I assume you will tackle holistically later.

**Comment by cuahsimarko** [comment]
> I am approving this PR with a comment that css styles can be managed as a separate css file rather than being embedding into html files and templates in vue files which I assume you will tackle holistically later.

Yes, I'm keeping them inline to track and make the job easier to consolidate the mess of overlapping css files, stale ones and includes and extends scopes all over the place.

**Comment by hydrocheck** [comment]
@cuahsimarko PR and regression testing suggests this is good. 👍

**Pull request merged**